app/api/v2/entities/user_with_full_info.rb

```diff
@@ -24,6 +25,16 @@ class UserWithFullInfo < API::V2::Entities::Base
         expose :phones, using: Entities::Phone
         expose :profile, using: Entities::Profile
         # activities, as sensitive and potentialy too big data should be queried separately
+
+        private
+
+        def secret
+          # csrf protection token
+          secret = SecureRandom.base64(32)
+
+          Rails.cache.write(object.uid, secret)
```

Comment:
we need to make sure this cache expire as the same date of the user session.

Reply:
I want to switch to redis session in the same patch and store the token at this same location

Reply:
This isn't a secret but a csrf token, also I would put it in entity is there a way we can show it only in login (session) controller ? Maybe

session {
  token: ...,
  expire: server time,
  user: entity
}

Also it would be nice if we protect sessions by IP
So we store session server side, among a token, uid, user-ip, user-agent

If token provided, user-ip, user-agent has changed we consider session invalid

Reply:
This is called a token not a secret